flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java

```diff
@@ -88,16 +88,18 @@
                                     + "we do not tolerance any checkpoint failure.");
 
     public static final ConfigOption<CheckpointConfig.ExternalizedCheckpointCleanup>
-            EXTERNALIZED_CHECKPOINT =
+            EXTERNALIZED_CHECKPOINT_CLEANUP =
                     ConfigOptions.key("execution.checkpointing.externalized-checkpoint-retention")
                             .enumType(CheckpointConfig.ExternalizedCheckpointCleanup.class)
-                            .noDefaultValue()
+                            .defaultValue(
+                                    CheckpointConfig.ExternalizedCheckpointCleanup
+                                            .NO_EXTERNALIZED_CHECKPOINTS)
                             .withDescription(
                                     Description.builder()
                                             .text(
                                                     "Externalized checkpoints write their meta data out to persistent storage and are not "
                                                             + "automatically cleaned up when the owning job fails or is suspended (terminating with job "
-                                                            + "status %s or %s. In this case, you have to manually clean up the checkpoint state, both the "
+                                                            + "status %s or %s). In this case, you have to manually clean up the checkpoint state, both the "
```

Comment:
As mentioned above, we should not describe the options here but let the enum do that.

Reply:
I am implementing `DescribedEnum` now to add some more description, but just for the record: The enum values are not mentioned here - `JobStatus` is a different enum...

Reply:
I left this description as it does not really describe the options, but rather the consequences for the users (you have to clean up leftover checkpoints yourself). Do you think it's too much overlap?